sky/spot/controller.py

```diff
@@ -24,6 +25,15 @@
 
 logger = sky_logging.init_logger(__name__)
 
+# We start process with 'spawn', because 'fork' results in weird
+# behaviors like ignoring SIGINT by the controller process when
+# it is started inside a Ray job. The previous fix was using and
+# catching SIGTERM instead, but recently I find this would crash the
+# process after ~10s, so this is clearly not a correct solution.
+# The correct solution uses 'spawn' to start a new process and
+# just send SIGINT for interruption.
+multiprocessing.set_start_method('spawn', force=True)
```

Comment:
Will calling this cause other multiprocessing we use to behave differently?

Also, according to document of multiprocessing ([ref](https://docs.python.org/3/library/multiprocessing.html#the-spawn-and-forkserver-start-methods)), we may want to place this line before L336 instead, instead in the range of `__name__ == '__main__'`?

Reply:
> This is a fascinating finding @suquark! Does that means if the teardown of the existing cluster takes longer than 10s, the spot cluster will be leaked as the controller process will be crashed? Can we try if adding time.sleep before the cluster termination will cause the leakage in the previous implementation? I will submit the review soon.

Yes, it leaks the spot instance in the new provisioner implementation. I can reproduce it (in the new provisioner) constantly by adding `time.sleep(60)` (actually less than 60s usually works, sometimes you just need to sleep 10s).